### DISCUSSION TITLE
Virtual File doesn't include embedded files when publish

### DISCUSSION THREAD

**Discussion by shawn-show** [abpframework/abp#12943] [Q&A]
I added a json file and made it to be embedded. And I also added the config on Module just followed the [document](https://docs.abp.io/en/abp/latest/Virtual-File-System#working-with-the-embedded-files)
```c#
Configure<AbpVirtualFileSystemOptions>(options =>
        {
            options.FileSets.AddEmbedded<DigitalTwinApplicationModule>();
           
        });
```
It worked when the solution running in VS. But when I published the solution to the server, it didn't work. I can not find the json file

**Comment by realLiangshiwei** [comment]
ABP all modules are configured the virtual file system, you can check it: https://github.com/abpframework/abp/blob/dev/modules/identity/src/Volo.Abp.Identity.Domain.Shared/Volo/Abp/Identity/AbpIdentityDomainSharedModule.cs#L23

**Reply by shawn-show** [reply]
No. I means I config the virtual file system in my module not abp module.

**Reply by realLiangshiwei** [reply]
I mean you can refer to it to configure your own module

**Comment by EngincanV** [comment]
Did you set the `GenerateEmbeddedFilesManifest` property to **true** and also use the [Microsoft.Extensions.FileProviders.Embedded](https://www.nuget.org/packages/Microsoft.Extensions.FileProviders.Embedded) package in your project that contains the embedded resource(s).?

```xml
<PropertyGroup>
   <GenerateEmbeddedFilesManifest>true</GenerateEmbeddedFilesManifest>
</PropertyGroup>

<ItemGroup>
    <PackageReference Include="Microsoft.Extensions.FileProviders.Embedded" Version="6.0.5" />
  </ItemGroup>
```

Also, please ensure you've marked your embedded resources correctly:

```xml
<ItemGroup>
  <EmbeddedResource Include="MyResources\**\*.*" />
  <Content Remove="MyResources\**\*.*" />
</ItemGroup>
```

**Reply by shawn-show** [reply]
Of course, I have added [Microsoft.Extensions.FileProviders.Embedded](https://www.nuget.org/packages/Microsoft.Extensions.FileProviders.Embedded) in my domain layer. And set GenerateEmbeddedFilesManifest to be true in my application layer which include my embedded json file.

**Reply by EngincanV** [reply]
If you have embedded files in your application layer, you should also add this package to your application layer.

> use the [Microsoft.Extensions.FileProviders.Embedded](https://www.nuget.org/packages/Microsoft.Extensions.FileProviders.Embedded) package in your project **that contains the embedded resource(s)**.

**Reply by maxrizwork** [reply]
I had a very similar issue recently when I published (passing from configuration Debug to Release) an app. 
abp version: 5.2.2
asp.net core webapp with a custom theme and a custom module with resources inside

I noticed I didn't have everywhere (where I'm using AbpVirtualFileSystemOptions) this package
Microsoft.Extensions.FileProviders.Embedded

By the way on web app project (ASP.NET Core) I prefer to NOT embed/include wwwroot folder, so I do NOT have this configuration:
`	
        <ItemGroup>
		<EmbeddedResource Include="wwwroot\**\*.*" />
		<Content Remove="wwwroot\**\*.*" />
	</ItemGroup>
`

**Reply by shawn-show** [reply]
> If you have embedded files in your application layer, you should also add this package to your application layer.
> 
> > use the [Microsoft.Extensions.FileProviders.Embedded](https://www.nuget.org/packages/Microsoft.Extensions.FileProviders.Embedded) package in your project **that contains the embedded resource(s)**.

Application layer is base on Domain layer, so I think install the package in Domain is enough
